migration/migration.go

```diff
@@ -31,31 +31,86 @@ var logger = loggo.GetLogger("juju.migration")
 type StateExporter interface {
 	// Export generates an abstract representation of a model.
 	Export(leaders map[string]string) (description.Model, error)
+	// ExportPartial the current model for the State optionally skipping
+	// aspects as defined by the ExportConfig.
+	ExportPartial(cfg state.ExportConfig) (description.Model, error)
 }
 
-// StateImporter describes the method needed to import a model
+// ExportMigrator represents a model migration that implements Export.
+type ExportMigrator struct {
+	stateExporter StateExporter
+	scope         modelmigration.Scope
+}
+
+func NewExportMigrator(stateExporter StateExporter, scope modelmigration.Scope) *ExportMigrator {
+	return &ExportMigrator{stateExporter, scope}
+}
+
+// ExportModelPartial partially serializes a model description from the
+// database contents, optionally skipping aspects as defined by the
+// ExportConfig.
+func (e *ExportMigrator) ExportModelPartial(ctx context.Context, cfg state.ExportConfig) (description.Model, error) {
+	model, err := e.stateExporter.ExportPartial(cfg)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return exportImpl(ctx, e.scope, model)
+}
+
+// ExportModel serializes a model description from the database contents.
+func (e *ExportMigrator) ExportModel(ctx context.Context, leaders map[string]string) (description.Model, error) {
+	model, err := e.stateExporter.Export(leaders)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	return exportImpl(ctx, e.scope, model)
+}
+
+func exportImpl(ctx context.Context, scope modelmigration.Scope, model description.Model) (description.Model, error) {
```

Comment:
Make this a member. Then you don't have to pass `scope`. It has no other usages, so should be encapsulated.
Just call it 'Export'.

Reply:
done